src/main/java/com/smartbear/postman/PostmanImporter.java

```diff
@@ -101,20 +101,15 @@ public PostmanImporter(TestCreator testCreator) {
 
     public WsdlProject importPostmanCollection(WorkspaceImpl workspace, String filePath) {
         WsdlProject project = null;
-        String postmanJson = null;
-        XProgressDialog collectionImportProgressDialog = UISupport.getDialogs().createProgressDialog("Import Collection",
-                0, "Importing the collection...", false);
-        PostmanImporterWorker worker = new PostmanImporterWorker(filePath);
-        try {
-            collectionImportProgressDialog.run(worker);
-        } catch (Exception e) {
-            e.printStackTrace();
-        }
-        postmanJson = worker.getPostmanJson();
+        String postmanJson = getPostmanImporterWorker(filePath).getPostmanJson();
         if (PostmanJsonUtil.seemsToBeJson(postmanJson)) {
             JSON json = new PostmanJsonUtil().parseTrimmedText(postmanJson);
             if (json instanceof JSONObject) {
                 PostmanCollection postmanCollection = PostmanCollectionFactory.getCollection((JSONObject) json);
+                if (postmanCollection == null) {
+                    logger.error("Error while extracting Postman collection");
+                    return null;
```

Comment:
Similar concern as in `PostmanCollectionFactory` - we get `null` from factory and pass `null` out from importer. The caller of this code will be responsible for proper handling and apply a null-check. Was it intended?

Reply:
I decided to return null as if we cannot extract collection the project cannot be created, similarly to the line 119 below where the SoapUIException is caught and null is returned. And I agree, showing a UI message is not a good decision. I suppose throwing an according exception will be better, as outside in ImportPostmanCollectionAction when exception is caught an error message is displayed in UI anyway